src/test/java/com/example/demo/MatcherControllerTest.java

```diff
@@ -0,0 +1,204 @@
+package com.example.demo;
+
+import com.example.demo.matcher.Matcher;
+import com.example.demo.matcher.MatcherController;
+import com.example.demo.matcher.models.*;
+import com.example.demo.matcher.services.OrderService;
+import com.example.demo.matcher.services.TradeService;
+import com.fasterxml.jackson.databind.ObjectMapper;
+
+import org.junit.jupiter.api.Test;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.autoconfigure.web.servlet.WebMvcTest;
+import org.springframework.boot.test.mock.mockito.MockBean;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.MediaType;
+import org.springframework.test.web.servlet.MockMvc;
+import org.springframework.test.web.servlet.MvcResult;
+import org.springframework.test.web.servlet.request.MockMvcRequestBuilders;
+
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+import java.util.UUID;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.mockito.Mockito.doReturn;
+
+@WebMvcTest(controllers = MatcherController.class)
+public class MatcherControllerTest {
+
+    @Autowired
+    private MockMvc mvc; // model view container, which handles the web stuff
+
+    @MockBean
```

Comment:
Is there a reason why this is a `@MockBean` and not `@Mock`?

Reply:
Because spring needs to know that they are beans and need to be injected into each other